packages/@storefront/core/src/core/url-beautifier/details/parser.ts

```diff
@@ -18,6 +18,10 @@ export class DetailsUrlParser extends UrlParser<UrlBeautifier.DetailsUrlState> {
     const title = path.shift();
     const id = path.pop();
     let variants = [];
+    const queryParams: { [key: string]: string } = {
+      ...( uri.query.area ? { area: uri.query.area } : {} ),
```

Comment:
I dawned on me that the query string parameters should probably be configurable, falling back to `area` and `collection` if the implementation doesn't provide custom keys. I think the PR should be updated to include this functionality.

Reply:
@jrmykolyn what do you mean by "configurable"?  Like, add an option in the config of whether or not to expect `area` and `collection` as query string params?